web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -112,27 +112,46 @@ function isR2L() {
     },
 
     make_multi: function() {
-        var $this = $(this),
-            all = $this.data('all'),
-            none = $this.data('none') || all,
-            allOpts = $this.data('all-options') || [],
-            extra = $this.data('extra'),
-            extraOpts = $this.data('extra-options') || [];
-
-            var presets = [{
-                name: all,
-                options: allOpts
-            }];
-
-            if (extra) {
-                presets.push({name: extra, options: extraOpts});
+        // A convenience wrapper around $.multiSelect() that translates HTML
+        // data-* attributes into settings for the multiSelect constructor.
+        var $select = $(this);
+        var settings = {};
+
+        if ( $select.data('none') ) {
+            settings.noneText = $select.data('none');
+        }
+
+        if ( $select.data('all') ) {
+            settings.allText = $select.data('all');
+
+            if ( ! settings.noneText ) {
+                settings.noneText = settings.allText;
             }
-        $this.multiSelect({
-            allText: all,
-            noneText: none,
-            positionMenuWithin: $('#side'),
-            presets: presets
-        });
+
+            if ( $select.data('all-options') ) {
```

Comment:
I think this changes the behaviour of the JS? Before there was always an `all` preset (even if there were no `all-options`, so it defaulted to `[]`) and now there isn't?

Reply:
Ah, interesting … what about if you don’t want an "all" option? (Like, I figured the `#area_ids` select in the body form didn’t need an "all" preset – maybe I'm wrong?)

Reply:
I agree, it doesn't. But that doesn't have a `data-all`? I mean if something has set a data-all but no all-options, you still want a preset then, as that's what it does currently. Wouldn't affect your new ones.

Reply:
Yeah, good point. Will add it back in now.